core/src/avm2/class.rs

```diff
@@ -280,6 +284,20 @@ impl<'gc> Class<'gc> {
         self.attributes = attributes;
     }
 
+    pub fn set_class_object(&mut self, class_object: ClassObject<'gc>) {
+        if self.class_object.is_some() {
```

Comment:
Can't you call `newclass` twice on the same `Class` and get two `ClassObject`s created pointing to the same class?

Reply:
Yes, but that would be an extremely cursed swf, and I'd prefer to get a panic report if that ever happens.

Reply:
I think the check should be removed, it's not that cursed of a case, and I feel like some SWF could end up tripping it. Besides, this should be unnecesary once the `ClassObject` refactor is completed.

Reply:
If this ever does happen, it's going to make the debug ui experience very confusing. If we don't want a panic, then I'd prefer to store a Vec, and show multiple ClassObjects in the ui somehow

Reply:
I've modified this to store a `Vec` and display a list in the debug ui